kernel/mm/pagetables.c

```diff
@@ -14,8 +14,6 @@
 #include <panic/panic.h>
 #include <string/string.h>
 
-pttentry *extract_cr3_base_address(pttentry cr3) __attribute__((alias("extract_pttentry_base_address")));
```

Comment:
I created the alias because I can envision a scenario where I might want to separate these two functions at some point, particularly once things progress where I'm putting page tables in locations that aren't identity-mapped and so I have to do some extra processing to turn the physical address that is in the page table into a virtual address that I can access from within the kernel process.  It might be necessary, it might not, but I created this alias to simplify maintenance if it ever does.

Reply:
ok cool.  If I created function "extract_cr3_base_address" that called extract_pttentry_base_address internally would that work for you.   I can't do the alias, just b/c the OS X clang compiler doesn't support aliases.

Reply:
@kurtmweber  changed the code back to two functions

Reply:
What do you think about something like this:

`#ifdef __GNUC__
`pttentry *extract_cr3_base_address(pttentry cr3) __attribute__((alias("extract_pttentry_base_address")));
#else
`pttentry *extract_cr3_base_address(pttentry entry){
`return extract_pttentry_base_address(entry);
`}
#endif

This avoids the overhead and extra size in the kernel (both of which are minimal, I realize) when the alias attribute is available, and further ensures that so long as both function calls are expected to do the same thing, if any changes need to be made they only need to be made in one place (eliminating the need to remember to make the changes in two places, at least until it's necessary to fully split them)?

Reply:
Change made